File: bench/Reference/file_readlines_large.erl
#!/usr/bin/env escript
%% Sum byte length of each line in the 50MB file.

main(_) ->
    {ok, Bin} = file:read_file("bench/data/large_text.txt"),
    Lines = binary:split(Bin, <<"\n">>, [global]),
    Total = lists:foldl(fun(L, Acc) -> Acc + byte_size(L) end, 0, Lines),
    io:format("~p~n", [Total]).

File: bench/Reference/FileReadlinesLarge.c
/* Stream 50MB file line-by-line, sum line lengths — idiomatic C */
#include <stdio.h>
#include <string.h>
int main(void) {
    FILE* f = fopen("bench/data/large_text.txt", "r");
    if (!f) return 1;
    char buf[8192];
    long total = 0;
    while (fgets(buf, sizeof(buf), f)) {
        size_t len = strlen(buf);
        if (len > 0 && buf[len-1] == '\n') len--;
        total += len;
    }
    fclose(f);
    printf("%ld\n", total);
    return 0;
}
